arbor/morph/region.cpp

```diff
@@ -0,0 +1,255 @@
+#include <set>
+#include <string>
+#include <vector>
+
+#include <arbor/morph/error.hpp>
+#include <arbor/morph/locset.hpp>
+#include <arbor/morph/primitives.hpp>
+#include <arbor/morph/region.hpp>
+
+#include "morph/em_morphology.hpp"
+#include "util/span.hpp"
+#include "util/strprintf.hpp"
+#include "util/range.hpp"
+#include "util/rangeutil.hpp"
+
+namespace arb {
+namespace reg {
+
+mcable_list merge(const mcable_list& lhs, const mcable_list& rhs) {
+    mcable_list v;
+    v.resize(lhs.size() + rhs.size());
+    std::merge(lhs.begin(), lhs.end(), rhs.begin(), rhs.end(), v.begin());
+    return v;
+}
+
+bool is_disjoint_union(const mcable& a, const mcable& b) {
+    if (a.branch!=b.branch) return true;
+    return a<b? a.dist_pos<b.prox_pos: b.dist_pos<a.prox_pos;
+}
+
+bool is_nonnull_intersection(const mcable& a, const mcable& b) {
+    if (a==b) return true; // handles special case of spherical branch
```

Comment:
It is! And no longer needed.
In older versions of this code `(cable 0 0 0)` was a perfectly acceptable way to refer to the whole sphere (because a sphere has inside and outside, but no ends).
Happily deleting that line keeps the unit tests happy.